commit 1e56279a49168f24b2b58e843ae92976f90a98a6
Author: Smita Koralahalli <Smita.KoralahalliChannabasappa@amd.com>
Date:   Thu Nov 4 16:58:42 2021 -0500

    x86/mce/inject: Set the valid bit in MCA_STATUS before error injection
    
    MCA handlers check the valid bit in each status register
    (MCA_STATUS[Val]) and continue processing the error only if the valid
    bit is set.
    
    Set the valid bit unconditionally in the corresponding MCA_STATUS
    register and correct any Val=0 injections made by the user as such
    errors will get ignored and such injections will be largely pointless.
    
    Signed-off-by: Smita Koralahalli <Smita.KoralahalliChannabasappa@amd.com>
    Signed-off-by: Borislav Petkov <bp@suse.de>
    Link: https://lore.kernel.org/r/20211104215846.254012-3-Smita.KoralahalliChannabasappa@amd.com

diff --git a/arch/x86/kernel/cpu/mce/inject.c b/arch/x86/kernel/cpu/mce/inject.c
index 725e8e4331a9..6eac840c64bb 100644
--- a/arch/x86/kernel/cpu/mce/inject.c
+++ b/arch/x86/kernel/cpu/mce/inject.c
@@ -503,6 +503,8 @@ static void do_inject(void)
 
 	i_mce.tsc = rdtsc_ordered();
 
+	i_mce.status |= MCI_STATUS_VAL;
+
 	if (i_mce.misc)
 		i_mce.status |= MCI_STATUS_MISCV;